diff -burN InsightToolkit-4.10.0.orig/Modules/ThirdParty/VNL/src/CMakeLists.txt InsightToolkit-4.10.0/Modules/ThirdParty/VNL/src/CMakeLists.txt
--- InsightToolkit-4.10.0.orig/Modules/ThirdParty/VNL/src/CMakeLists.txt	2016-11-23 15:59:11.409392257 +0100
+++ InsightToolkit-4.10.0/Modules/ThirdParty/VNL/src/CMakeLists.txt	2016-11-23 16:26:47.185472392 +0100
@@ -18,7 +18,7 @@
 # Retrive the variable type to CACHE.
 set(BUILD_EXAMPLES ${BUILD_EXAMPLES} CACHE BOOL "Build the examples from the ITK Software Guide." FORCE)
 
-foreach(lib itkvcl itkv3p_netlib itktestlib itkvnl itkvnl_algo itknetlib)
+foreach(lib itkvcl itkv3p_netlib itkvnl itkvnl_algo itknetlib)
   itk_module_target(${lib} NO_INSTALL)
 endforeach()
 
diff -burN InsightToolkit-4.10.0.orig/Modules/ThirdParty/ZLIB/CMakeLists.txt InsightToolkit-4.10.0/Modules/ThirdParty/ZLIB/CMakeLists.txt
--- InsightToolkit-4.10.0.orig/Modules/ThirdParty/ZLIB/CMakeLists.txt	2016-11-23 15:59:11.421392257 +0100
+++ InsightToolkit-4.10.0/Modules/ThirdParty/ZLIB/CMakeLists.txt	2016-11-23 16:26:35.889471845 +0100
@@ -6,14 +6,12 @@
 
 if(ITK_USE_SYSTEM_ZLIB)
   find_package(ZLIB REQUIRED)
-  set(ITKZLIB_INCLUDE_DIRS
-    ${ITKZLIB_BINARY_DIR}/src
-    )
-  set(ITKZLIB_SYSTEM_INCLUDE_DIRS
-    ${ZLIB_INCLUDE_DIRS}
-    )
-  set(ITKZLIB_LIBRARIES "${ZLIB_LIBRARIES}")
   set(ITKZLIB_NO_SRC 1)
+  set(ITKZLIB_EXPORT_CODE_BUILD
+"set(ITKZLIB_INCLUDE_DIRS \"${ITKZLIB_BINARY_DIR}/src;${ZLIB_INCLUDE_DIRS}\" ) 
+set(ITKZLIB_LIBRARIES \"${ZLIB_LIBRARIES}\")  
+")
+set(ITKZLIB_EXPORT_CODE_INSTALL "find_package(ZLIB REQUIRED)")
 else()
   set(ITKZLIB_INCLUDE_DIRS
     ${ITKZLIB_BINARY_DIR}/src
