diff -burN ITK-5.0b01.orig/CMake/FindFFTW.cmake ITK-5.0b01/CMake/FindFFTW.cmake
--- ITK-5.0b01.orig/CMake/FindFFTW.cmake	Wed Oct  3 11:56:29 2018
+++ ITK-5.0b01/CMake/FindFFTW.cmake	Wed Oct  3 11:58:11 2018
@@ -56,7 +56,7 @@
     )
   endif()
 
-  find_path(FFTW_INCLUDE_PATH fftw3.h ${FFTW_INC_SEARCHPATH})
+  find_path(FFTW_INCLUDE_PATH fftw3.h NO_DEFAULT_PATH)
 
   if(FFTW_INCLUDE_PATH)
     file(TO_CMAKE_PATH "${FFTW_INCLUDE_PATH}" FFTW_INCLUDE_PATH)
@@ -145,14 +145,11 @@
     get_filename_component(FFTW_INSTALL_BASE_PATH ${FFTW_INCLUDE_PATH} PATH)
     set(FFTW_LIB_SEARCHPATH
       ${FFTW_INSTALL_BASE_PATH}/lib
-      ${FFTW_INSTALL_BASE_PATH}/lib64
-      /usr/lib/fftw
-      /usr/local/lib/fftw
     )
     if(ITK_USE_FFTWD)
       mark_as_advanced(FFTWD_LIB)
-      find_library(FFTWD_LIB fftw3 ${FFTW_LIB_SEARCHPATH}) #Double Precision Lib
-      find_library(FFTWD_THREADS_LIB fftw3_threads ${FFTW_LIB_SEARCHPATH}) #Double Precision Lib only if compiled with threads support
+      find_library(FFTWD_LIB fftw3 ${FFTW_LIB_SEARCHPATH}  NO_DEFAULT_PATH) #Double Precision Lib
+      find_library(FFTWD_THREADS_LIB fftw3_threads ${FFTW_LIB_SEARCHPATH}  NO_DEFAULT_PATH) #Double Precision Lib only if compiled with threads support
 
       if(FFTWD_LIB)
         set(FFTWD_FOUND 1)
@@ -167,8 +164,8 @@
 
     if(ITK_USE_FFTWF)
       mark_as_advanced(FFTWF_LIB)
-      find_library(FFTWF_LIB fftw3f ${FFTW_LIB_SEARCHPATH}) #Single Precision Lib
-      find_library(FFTWF_THREADS_LIB fftw3f_threads ${FFTW_LIB_SEARCHPATH}) #Single Precision Lib only if compiled with threads support
+      find_library(FFTWF_LIB fftw3f ${FFTW_LIB_SEARCHPATH}  NO_DEFAULT_PATH) #Single Precision Lib
+      find_library(FFTWF_THREADS_LIB fftw3f_threads ${FFTW_LIB_SEARCHPATH}  NO_DEFAULT_PATH) #Single Precision Lib only if compiled with threads support
 
       if(FFTWF_LIB)
         set(FFTWF_FOUND 1)
