--- InsightToolkit-4.10.0.orig/Modules/Core/Common/include/itkImageAlgorithm.h	2020-07-24 03:47:25.000000000 +0200
+++ InsightToolkit-4.10.0/Modules/Core/Common/include/itkImageAlgorithm.h	2021-02-24 08:22:40.193765449 +0100
@@ -190,7 +190,6 @@
   /** Unary functor just for static_cast operator */
   template<typename TInputType, typename TOutputType>
   struct StaticCast
-    : public std::unary_function<TInputType,TOutputType>
     {
     TOutputType operator() (const TInputType i) { return static_cast<TOutputType>(i); }
     };
