diff -burN netcdf-4.9.3-orig/CMakeLists.txt netcdf-4.9.3/CMakeLists.txt
--- netcdf-4.9.3-orig/CMakeLists.txt  2025-11-18 17:02:38.591775473 +0100
+++ netcdf-4.9.3/CMakeLists.txt       2025-11-18 17:19:20.021291946 +0100
@@ -134,6 +134,7 @@
 
 # For CMAKE_INSTALL_LIBDIR
 include(GNUInstallDirs)
+set(CMAKE_INSTALL_LIBDIR lib)
 
 if(MSVC)
   set(GLOBAL PROPERTY USE_FOLDERS ON)
diff -burN netcdf-4.9.3-orig/cmake/dependencies.cmake netcdf-4.9.3/cmake/dependencies.cmake
--- netcdf-4.9.3-orig/cmake/dependencies.cmake.orig        2025-11-18 17:45:11.053175621 +0100
+++ netcdf-4.9.3/cmake/dependencies.cmake     2025-11-18 17:45:23.760302869 +0100
@@ -29,12 +29,12 @@
   )
   endif()
 
-  find_library(HDF4_DF_LIB NAMES df libdf hdf)
+  find_library(HDF4_DF_LIB NAMES dfalt df libdf hdf)
   if(NOT HDF4_DF_LIB)
     message(FATAL_ERROR "Can't find or link to the hdf4 df library.")
   endif()
 
-  find_library(HDF4_MFHDF_LIB NAMES mfhdf libmfhdf)
+  find_library(HDF4_MFHDF_LIB NAMES mfhdfalt mfhdf libmfhdf)
   if(NOT HDF4_MFHDF_LIB)
     message(FATAL_ERROR "Can't find or link to the hdf4 mfhdf library.")
   endif()
