diff -burN libgeotiff-1.4.1.orig/configure libgeotiff-1.4.1/configure
--- libgeotiff-1.4.1.orig/configure	2016-04-15 17:57:22.000000000 +0200
+++ libgeotiff-1.4.1/configure	2016-04-15 18:00:56.000000000 +0200
@@ -16101,7 +16101,7 @@
 
     PROJ_CONFIG="yes"
 
-    elif test \( -f "$PROJ_HOME/lib/libproj.a" -o -f "$PROJ_HOME/lib/libproj.so" \) -a -f "$PROJ_HOME/include/proj_api.h"; then
+    elif test \( -f "$PROJ_HOME/lib/libproj.a" -o -f "$PROJ_HOME/lib/libproj.so" -o -f "$PROJ_HOME/lib/libproj.dylib" \) -a -f "$PROJ_HOME/include/proj_api.h"; then
     PROJ_LIB="-L$PROJ_HOME/lib -lproj"
     PROJ_INC=$PROJ_HOME/include
 
@@ -21017,5 +21017,3 @@
 
 
 echo ""
-
-
