diff -burN libgeotiff-1.5.1.orig/CMakeLists.txt libgeotiff-1.5.1/CMakeLists.txt
--- libgeotiff-1.5.1.orig/CMakeLists.txt	2019-03-29 21:09:12.000000000 +0100
+++ libgeotiff-1.5.1/CMakeLists.txt	2019-11-08 11:17:08.520146442 +0100
@@ -345,12 +345,12 @@
       VERSION ${LINK_VERSION}
       SOVERSION ${LINK_SOVERSION}
       CLEAN_DIRECT_OUTPUT 1 )
-   if (APPLE)
-      set_target_properties(
-        ${GEOTIFF_LIBRARY_TARGET}
-        PROPERTIES
-        INSTALL_NAME_DIR "${CMAKE_INSTALL_PREFIX}/lib")
-   endif()
+   # if (APPLE)
+   #    set_target_properties(
+   #      ${GEOTIFF_LIBRARY_TARGET}
+   #      PROPERTIES
+   #      INSTALL_NAME_DIR "${CMAKE_INSTALL_PREFIX}/lib")
+   # endif()
 
 ELSE(UNIX)
 # Default:
