diff -burN libgeotiff-1.4.2.orig/CMakeLists.txt libgeotiff-1.4.2/CMakeLists.txt
--- libgeotiff-1.4.2.orig/CMakeLists.txt	2019-05-27 11:41:34.203242519 +0200
+++ libgeotiff-1.4.2/CMakeLists.txt	2019-05-27 11:41:44.735282436 +0200
@@ -424,7 +424,7 @@
 if (MSVC OR CMAKE_CONFIGURATION_TYPES)
   # For multi-config systems and for Visual Studio, the debug versions
   # of the libraries have a _d suffix.
-  set (CMAKE_DEBUG_POSTFIX _d)
+#  set (CMAKE_DEBUG_POSTFIX _d)
 endif ()
 
 ADD_LIBRARY(${GEOTIFF_ARCHIVE_TARGET} STATIC
