diff -burN libgeotiff-1.7.1.orig/CMakeLists.txt libgeotiff-1.7.1/CMakeLists.txt
--- libgeotiff-1.7.1.orig/CMakeLists.txt	2022-03-10 09:32:14.000000000 +0100
+++ libgeotiff-1.7.1/CMakeLists.txt	2022-06-29 16:32:45.867700159 +0200
@@ -319,12 +319,12 @@
       VERSION ${LINK_VERSION}
       SOVERSION ${LINK_SOVERSION}
       CLEAN_DIRECT_OUTPUT 1 )
-   if (APPLE)
-      set_target_properties(
-        ${GEOTIFF_LIBRARY_TARGET}
-        PROPERTIES
-        INSTALL_NAME_DIR "${CMAKE_INSTALL_PREFIX}/${GEOTIFF_LIB_DIR}")
-   endif()
+   # if (APPLE)
+   #    set_target_properties(
+   #      ${GEOTIFF_LIBRARY_TARGET}
+   #      PROPERTIES
+   #      INSTALL_NAME_DIR "${CMAKE_INSTALL_PREFIX}/${GEOTIFF_LIB_DIR}")
+   # endif()
 endif()
 
 SET_TARGET_PROPERTIES(${GEOTIFF_LIBRARY_TARGET} PROPERTIES
