diff -burN hdf5-1.10.1.orig/config/cmake_ext_mod/HDFMacros.cmake hdf5-1.10.1/config/cmake_ext_mod/HDFMacros.cmake
--- hdf5-1.10.1.orig/config/cmake_ext_mod/HDFMacros.cmake	2019-05-27 14:52:13.908329096 +0200
+++ hdf5-1.10.1/config/cmake_ext_mod/HDFMacros.cmake	2019-05-27 14:52:53.716811437 +0200
@@ -91,18 +91,18 @@
   if (${libtype} MATCHES "SHARED")
     if (WIN32)
       set (LIB_RELEASE_NAME "${libname}")
-      set (LIB_DEBUG_NAME "${libname}_D")
+      set (LIB_DEBUG_NAME "${libname}")
     else ()
       set (LIB_RELEASE_NAME "${libname}")
-      set (LIB_DEBUG_NAME "${libname}_debug")
+      set (LIB_DEBUG_NAME "${libname}")
     endif ()
   else ()
     if (WIN32)
       set (LIB_RELEASE_NAME "lib${libname}")
-      set (LIB_DEBUG_NAME "lib${libname}_D")
+      set (LIB_DEBUG_NAME "lib${libname}")
     else ()
       set (LIB_RELEASE_NAME "${libname}")
-      set (LIB_DEBUG_NAME "${libname}_debug")
+      set (LIB_DEBUG_NAME "${libname}")
     endif ()
   endif ()
 
