diff -burN ossim-1.8.20-3.orig/ossim/CMakeLists.txt ossim-1.8.20-3/ossim/CMakeLists.txt
--- ossim-1.8.20-3.orig/ossim/CMakeLists.txt	2016-04-19 12:29:59.000000000 +0200
+++ ossim-1.8.20-3/ossim/CMakeLists.txt	2016-04-20 09:21:26.000000000 +0200
@@ -160,7 +160,7 @@
 #---
 TODAYS_DATE(OSSIM_BUILD_DATE)
 SET(OSSIM_SVN_REVISION_NUMBER "UNKNOWN")
-GET_SVN_REVISION()
+#GET_SVN_REVISION()
 if ( NOT ${Project_WC_REVISION} EQUAL 0 )
    set( OSSIM_SVN_REVISION_NUMBER ${Project_WC_REVISION} )
 endif()
