--- ossim-1.8.20-3.orig/ossim_package_support/cmake/CMakeModules/FindFreetype.cmake	Fri Jun 29 21:54:24 2012
+++ ossim-1.8.20-3/ossim_package_support/cmake/CMakeModules/FindFreetype.cmake	Mon Sep  2 10:13:12 2019
@@ -44,26 +44,7 @@
 FIND_PATH(FREETYPE_INCLUDE_DIR_ft2build ft2build.h 
-  HINTS
-  $ENV{FREETYPE_DIR}
-  PATH_SUFFIXES include
-  PATHS
-  /usr/local/X11R6/include
-  /usr/local/X11/include
-  /usr/X11/include
-  /sw/include
-  /opt/local/include
-  /usr/freeware/include
+PATH_SUFFIXES include include/freetype2
 )
 
-FIND_PATH(FREETYPE_INCLUDE_DIR_freetype2 freetype/config/ftheader.h 
-  HINTS
-  $ENV{FREETYPE_DIR}/include/freetype2
-  PATHS
-  /usr/local/X11R6/include
-  /usr/local/X11/include
-  /usr/X11/include
-  /sw/include
-  /opt/local/include
-  /usr/freeware/include
-  PATH_SUFFIXES freetype2
-)
+FIND_PATH(FREETYPE_INCLUDE_DIR_freetype2 config/ftheader.h 
+  PATH_SUFFIXES include include/freetype2)
 
@@ -71,11 +52,2 @@
   NAMES freetype libfreetype freetype219 freetype239
-  HINTS
-  $ENV{FREETYPE_DIR}
-  PATH_SUFFIXES lib64 lib
-  PATHS
-  /usr/local/X11R6
-  /usr/local/X11
-  /usr/X11
-  /sw
-  /usr/freeware
 )
--- ossim-1.8.20-3.orig/ossim_package_support/cmake/CMakeModules/FindGEOTIFF.cmake	Tue Jan 20 19:25:11 2015
+++ ossim-1.8.20-3/ossim_package_support/cmake/CMakeModules/FindGEOTIFF.cmake	Mon Sep  2 10:11:53 2019
@@ -17,3 +17,3 @@
 
-SET(GEOTIFF_NAMES ${GEOTIFF_NAMES} geotiff_i geotiff libgeotiff_i libgeotiff)
+SET(GEOTIFF_NAMES ${GEOTIFF_NAMES} geotiff_d_i geotiff_i geotiff libgeotiff_i libgeotiff)
 FIND_LIBRARY(GEOTIFF_LIBRARY 
