diff -burN qt-everywhere-opensource-src-4.8.7.orig/src/plugins/graphicssystems/graphicssystems.pro qt-everywhere-opensource-src-4.8.7/src/plugins/graphicssystems/graphicssystems.pro
--- qt-everywhere-opensource-src-4.8.7.orig/src/plugins/graphicssystems/graphicssystems.pro	2017-08-17 11:59:00.122963214 +0200
+++ qt-everywhere-opensource-src-4.8.7/src/plugins/graphicssystems/graphicssystems.pro	2017-08-17 12:00:04.699387068 +0200
@@ -1,5 +1,4 @@
 TEMPLATE = subdirs
-SUBDIRS += trace
 !wince*:contains(QT_CONFIG, opengl):SUBDIRS += opengl
 contains(QT_CONFIG, openvg):contains(QT_CONFIG, egl) {
     SUBDIRS += openvg
diff -burN qt-everywhere-opensource-src-4.8.7.orig/src/plugins/plugins.pro qt-everywhere-opensource-src-4.8.7/src/plugins/plugins.pro
--- qt-everywhere-opensource-src-4.8.7.orig/src/plugins/plugins.pro	2017-08-17 11:59:00.138963317 +0200
+++ qt-everywhere-opensource-src-4.8.7/src/plugins/plugins.pro	2017-08-17 12:02:45.716444268 +0200
@@ -1,16 +1,16 @@
 TEMPLATE = subdirs
 
-SUBDIRS	*= sqldrivers script bearer
-unix:!symbian {
-        contains(QT_CONFIG,iconv)|contains(QT_CONFIG,gnu-libiconv)|contains(QT_CONFIG,sun-libiconv):SUBDIRS *= codecs
-} else {
-        SUBDIRS *= codecs
-}
+# SUBDIRS	*= sqldrivers
+# unix:!symbian {
+#        contains(QT_CONFIG,iconv)|contains(QT_CONFIG,gnu-libiconv)|contains(QT_CONFIG,sun-libiconv):SUBDIRS *= codecs
+# } else {
+#         SUBDIRS *= codecs
+# }
 !contains(QT_CONFIG, no-gui): SUBDIRS *= imageformats iconengines
 !embedded:!qpa:!contains(QT_CONFIG, no-gui):SUBDIRS *= graphicssystems
 embedded:SUBDIRS *=  gfxdrivers decorations mousedrivers kbddrivers
-!win32:!embedded:!mac:!symbian:!contains(QT_CONFIG, no-gui):SUBDIRS *= inputmethods
-!symbian:!contains(QT_CONFIG, no-gui):SUBDIRS += accessible
+# !win32:!embedded:!mac:!symbian:!contains(QT_CONFIG, no-gui):SUBDIRS *= inputmethods
+# !symbian:!contains(QT_CONFIG, no-gui):SUBDIRS += accessible
 contains(QT_CONFIG, phonon): SUBDIRS *= phonon
 qpa:SUBDIRS += platforms
 contains(QT_CONFIG, declarative): SUBDIRS *= qmltooling
diff -burN qt-everywhere-opensource-src-4.8.7.orig/src/src.pro qt-everywhere-opensource-src-4.8.7/src/src.pro
--- qt-everywhere-opensource-src-4.8.7.orig/src/src.pro	2017-08-17 11:59:00.870968087 +0200
+++ qt-everywhere-opensource-src-4.8.7/src/src.pro	2017-08-17 12:00:51.991697564 +0200
@@ -4,7 +4,7 @@
 unset(SRC_SUBDIRS)
 win32:SRC_SUBDIRS += src_winmain
 symbian:SRC_SUBDIRS += src_s60main
-SRC_SUBDIRS += src_corelib src_xml src_network src_sql src_testlib
+SRC_SUBDIRS += src_corelib src_xml
 nacl: SRC_SUBDIRS -= src_network src_testlib
 !symbian:contains(QT_CONFIG, dbus):SRC_SUBDIRS += src_dbus
 !contains(QT_CONFIG, no-gui): SRC_SUBDIRS += src_gui
