diff -burN geos-3.6.1.orig/cmake/modules/GenerateSourceGroups.cmake geos-3.6.1/cmake/modules/GenerateSourceGroups.cmake
--- geos-3.6.1.orig/cmake/modules/GenerateSourceGroups.cmake	1970-01-01 01:00:00.000000000 +0100
+++ geos-3.6.1/cmake/modules/GenerateSourceGroups.cmake	2017-06-19 11:19:32.143054641 +0200
@@ -0,0 +1,17 @@
+#
+# Macro generates tree of IDE source groups based on folders structure
+# Source: http://www.cmake.org/pipermail/cmake/2013-November/056332.html
+# 
+macro(GenerateSourceGroups curdir)
+  file(GLOB children RELATIVE ${PROJECT_SOURCE_DIR}/${curdir} ${PROJECT_SOURCE_DIR}/${curdir}/*)
+  foreach(child ${children})
+    if(IS_DIRECTORY ${PROJECT_SOURCE_DIR}/${curdir}/${child})
+      GenerateSourceGroups(${curdir}/${child})
+    else()
+      string(REPLACE "/" "\\" groupname ${curdir})
+      # I would like to call the src root folder in a different name, only in visual studio (not mandatory requirement)
+	  string(REPLACE "src" "Source Files" groupname ${groupname})
+      source_group(${groupname} FILES ${PROJECT_SOURCE_DIR}/${curdir}/${child})
+    endif()
+  endforeach()
+endmacro()
\ Pas de fin de ligne à la fin du fichier
