diff -burN geos-3.6.0dev.orig/CMakeLists.txt geos-3.6.0dev/CMakeLists.txt
--- geos-3.6.0dev.orig/CMakeLists.txt	2016-06-16 14:53:39.042188675 +0200
+++ geos-3.6.0dev/CMakeLists.txt	2016-06-16 15:04:21.562183651 +0200
@@ -254,7 +254,7 @@
 ENDMACRO(GET_SVN_REVISION)
 
 # Determine SVN/Git revision
-set(GEOS_BUILD_PACKAGED TRUE)
+set(GEOS_BUILD_PACKAGED FALSE)
 if(EXISTS "${PROJECT_SOURCE_DIR}/.svn")
   set(GEOS_BUILD_PACKAGED FALSE)
   GET_SVN_REVISION()
diff -burN geos-3.6.0dev.orig/src/CMakeLists.txt geos-3.6.0dev/src/CMakeLists.txt
--- geos-3.6.0dev.orig/src/CMakeLists.txt	2016-06-16 14:53:39.062188675 +0200
+++ geos-3.6.0dev/src/CMakeLists.txt	2016-06-16 15:31:00.450171151 +0200
@@ -62,7 +62,7 @@
 else()
 
   add_library(geos SHARED ${geos_SOURCES} ${geos_ALL_HEADERS})
-  add_library(geos-static STATIC ${geos_SOURCES} ${geos_ALL_HEADERS})
+#  add_library(geos-static STATIC ${geos_SOURCES} ${geos_ALL_HEADERS})
 
   set_target_properties(geos
     PROPERTIES
@@ -70,11 +70,11 @@
     VERSION ${VERSION}
     CLEAN_DIRECT_OUTPUT 1)
 
-  set_target_properties(geos-static
-    PROPERTIES
-    OUTPUT_NAME "geos"
-    PREFIX "lib"
-    CLEAN_DIRECT_OUTPUT 1)
+  # set_target_properties(geos-static
+  #   PROPERTIES
+  #   OUTPUT_NAME "geos"
+  #   PREFIX "lib"
+  #   CLEAN_DIRECT_OUTPUT 1)
 
 endif()
 
@@ -90,7 +90,7 @@
 #################################################################################
 
 if(NOT GEOS_ENABLE_MACOSX_FRAMEWORK)
-  install(TARGETS geos geos-static
+  install(TARGETS geos
     RUNTIME DESTINATION bin
     LIBRARY DESTINATION lib
     ARCHIVE DESTINATION lib)
