diff -burN glew-1.13.0.orig/build/cmake/CMakeLists.txt glew-1.13.0/build/cmake/CMakeLists.txt
--- glew-1.13.0.orig/build/cmake/CMakeLists.txt	Thu Feb  2 11:07:22 2017
+++ glew-1.13.0/build/cmake/CMakeLists.txt	Thu Feb  2 11:08:28 2017
@@ -10,8 +10,9 @@
   cmake_policy (SET CMP0003 NEW)
 endif()
 
-set(CMAKE_DEBUG_POSTFIX d)
-
+############# OTB ############# 
+#set(CMAKE_DEBUG_POSTFIX d)
+############# OTB #############
 option (BUILD_UTILS "utilities" ON)
 
 set (GLEW_DIR ${CMAKE_CURRENT_SOURCE_DIR}/../..)
