diff -burN curl-7.40.0.orig/winbuild/Makefile.vc curl-7.40.0/winbuild/Makefile.vc
--- curl-7.40.0.orig/winbuild/Makefile.vc	Fri Jul  8 16:09:05 2016
+++ curl-7.40.0/winbuild/Makefile.vc	Fri Jul  8 16:10:02 2016
@@ -123,7 +123,7 @@
 SSH2     = static
 !ENDIF
 
-CONFIG_NAME_LIB = $(CONFIG_NAME_LIB)-vc$(VC)-$(MACHINE)
+CONFIG_NAME_LIB = $(CONFIG_NAME_LIB)-vc$(VC)
 
 !IF "$(DEBUG)"=="yes"
 CONFIG_NAME_LIB = $(CONFIG_NAME_LIB)-debug
