diff -burN hdf-4.2.13.orig/config/cmake_ext_mod/HDFMacros.cmake hdf-4.2.13/config/cmake_ext_mod/HDFMacros.cmake
--- hdf-4.2.13.orig/config/cmake_ext_mod/HDFMacros.cmake	2017-06-24 23:00:04.000000000 +0200
+++ hdf-4.2.13/config/cmake_ext_mod/HDFMacros.cmake	2019-05-27 14:42:59.257664634 +0200
@@ -80,18 +80,18 @@
   if (${libtype} MATCHES "SHARED")
     if (WIN32)
       set (LIB_RELEASE_NAME "${libname}")
-      set (LIB_DEBUG_NAME "${libname}_D")
+      set (LIB_DEBUG_NAME "${libname}")
     else ()
       set (LIB_RELEASE_NAME "${libname}")
-      set (LIB_DEBUG_NAME "${libname}_debug")
+      set (LIB_DEBUG_NAME "${libname}")
     endif ()
   else ()
     if (WIN32)
       set (LIB_RELEASE_NAME "lib${libname}")
-      set (LIB_DEBUG_NAME "lib${libname}_D")
+      set (LIB_DEBUG_NAME "lib${libname}")
     else ()
       set (LIB_RELEASE_NAME "${libname}")
-      set (LIB_DEBUG_NAME "${libname}_debug")
+      set (LIB_DEBUG_NAME "${libname}")
     endif ()
   endif ()
 
