--- poppler-0.31.0.orig/poppler/poppler-config.h.cmake	Sun Jan  4 23:10:57 2015
+++ poppler-0.31.0/poppler/poppler-config.h.cmake	Thu Jul 18 12:21:26 2019
@@ -27,3 +27,3 @@
 #include <stdio.h>
-
+#include <algorithm>
 // We duplicate some of the config.h #define's here since they are
--- poppler-0.31.0.orig/CMakeLists.txt	Thu Feb  5 19:37:11 2015
+++ poppler-0.31.0/CMakeLists.txt	Thu Jul 18 12:31:49 2019
@@ -650,3 +650,3 @@
 endif(ENABLE_GLIB)
-add_subdirectory(test)
+# add_subdirectory(test)
 if(QT4_FOUND)
--- poppler-0.31.0.orig/config.h.cmake	Sun Sep 28 17:19:59 2014
+++ poppler-0.31.0/config.h.cmake	Thu Jul 18 11:53:22 2019
@@ -218,6 +218,2 @@
 
-/* MS has defined snprintf as deprecated */
-#ifdef _MSC_VER
-#define snprintf _snprintf
-#endif
 
@@ -231 +227,3 @@
 /* TODO This is wrong, port if needed #undef _LARGE_FILES */
+
+#include <algorithm>
