diff -burN opencv-3.4.1.orig/CMakeLists.txt opencv-3.4.1/CMakeLists.txt
--- opencv-3.4.1.orig/CMakeLists.txt	2019-05-27 11:56:25.490796576 +0200
+++ opencv-3.4.1/CMakeLists.txt	2019-05-27 11:57:26.447077196 +0200
@@ -491,8 +491,8 @@
 
 if(WIN32)
   # Postfix of DLLs:
-  set(OPENCV_DLLVERSION "${OPENCV_VERSION_MAJOR}${OPENCV_VERSION_MINOR}${OPENCV_VERSION_PATCH}")
-  set(OPENCV_DEBUG_POSTFIX d)
+  set(OPENCV_DLLVERSION "")#${OPENCV_VERSION_MAJOR}${OPENCV_VERSION_MINOR}${OPENCV_VERSION_PATCH}")
+  set(OPENCV_DEBUG_POSTFIX "")
 else()
   # Postfix of so's:
   set(OPENCV_DLLVERSION "")
