diff -burN libjpeg-turbo-1.4.1.orig/CMakeLists.txt libjpeg-turbo-1.4.1/CMakeLists.txt
--- libjpeg-turbo-1.4.1.orig/CMakeLists.txt	2016-09-06 09:37:12.107984277 +0200
+++ libjpeg-turbo-1.4.1/CMakeLists.txt	2016-09-06 11:27:25.663932571 +0200
@@ -167,6 +167,7 @@
 
 message(STATUS "Install directory = ${CMAKE_INSTALL_PREFIX}")
 
+set(BUILD "04/02/2016")
 configure_file(win/jconfig.h.in jconfig.h)
 configure_file(win/jconfigint.h.in jconfigint.h)
 
