diff -burN libjpeg-turbo-1.4.1.orig/CMakeLists.txt libjpeg-turbo-1.4.1/CMakeLists.txt
--- libjpeg-turbo-1.4.1.orig/CMakeLists.txt	2016-05-19 08:20:33.000000000 +0200
+++ libjpeg-turbo-1.4.1/CMakeLists.txt	2016-05-19 09:24:28.000000000 +0200
@@ -167,6 +167,11 @@
 
 message(STATUS "Install directory = ${CMAKE_INSTALL_PREFIX}")
 
+
+include(CheckTypeSize)
+check_type_size("size_t" JPEG_SIZEOF_SIZE_T)
+set(CMAKE_C_FLAGS ${CMAKE_C_FLAGS} -DSIZEOF_SIZE_T=${JPEG_SIZEOF_SIZE_T})
+
 configure_file(win/jconfig.h.in jconfig.h)
 configure_file(win/jconfigint.h.in jconfigint.h)
 
@@ -287,9 +292,9 @@
 endif()
 
 if(WITH_12BIT)
-  set(COMPILE_FLAGS "-DGIF_SUPPORTED -DPPM_SUPPORTED -DUSE_SETMODE")
+  set(COMPILE_FLAGS "-DGIF_SUPPORTED -DPPM_SUPPORTED")
 else()
-  set(COMPILE_FLAGS "-DBMP_SUPPORTED -DGIF_SUPPORTED -DPPM_SUPPORTED -DTARGA_SUPPORTED -DUSE_SETMODE")
+  set(COMPILE_FLAGS "-DBMP_SUPPORTED -DGIF_SUPPORTED -DPPM_SUPPORTED -DTARGA_SUPPORTED")
 	set(CJPEG_BMP_SOURCES rdbmp.c rdtarga.c)
 	set(DJPEG_BMP_SOURCES wrbmp.c wrtarga.c)
 endif()
diff -burN libjpeg-turbo-1.4.1.orig/win/jconfig.h.in libjpeg-turbo-1.4.1/win/jconfig.h.in
--- libjpeg-turbo-1.4.1.orig/win/jconfig.h.in	2016-05-19 08:20:33.000000000 +0200
+++ libjpeg-turbo-1.4.1/win/jconfig.h.in	2016-05-19 09:26:12.000000000 +0200
@@ -48,3 +48,5 @@
 #undef RIGHT_SHIFT_IS_UNSIGNED
 
 #endif /* JPEG_INTERNALS */
+
+#undef USE_SETMODE
