diff -burN proj-6.2.1_orig/CMakeLists.txt proj-6.2.1/CMakeLists.txt
--- proj-6.2.1_orig/CMakeLists.txt	2019-10-28 18:49:32.000000000 +0100
+++ proj-6.2.1/CMakeLists.txt	2019-11-08 10:52:28.948109756 +0100
@@ -98,6 +98,7 @@
 proj_version(MAJOR 6 MINOR 2 PATCH 1)
 set(PROJ_API_VERSION "17")
 set(PROJ_BUILD_VERSION "17.1.2")
+option(VERSIONED_OUTPUT "suffix version into output name of binaries." ON)
 
 ################################################################################
 # Build features and variants

diff -burN proj-6.2.1_orig/src/lib_proj.cmake proj-6.2.1_orig/src/lib_proj.cmake
--- proj-6.2.1_orig/src/lib_proj.cmake	2019-10-28 17:49:41.000000000 +0100
+++ proj-6.2.1_orig/src/lib_proj.cmake	2019-11-12 12:00:09.696226412 +0100
@@ -401,11 +401,18 @@
 endif()
 
 if(WIN32)
+  if(VERSIONED_OUTPUT)
   set_target_properties(${PROJ_CORE_TARGET}
     PROPERTIES
     VERSION "${${PROJECT_INTERN_NAME}_BUILD_VERSION}"
     OUTPUT_NAME "${PROJ_CORE_TARGET_OUTPUT_NAME}"
     CLEAN_DIRECT_OUTPUT 1)
+  else()
+      set_target_properties(${PROJ_CORE_TARGET}
+      PROPERTIES
+      VERSION "${${PROJECT_INTERN_NAME}_BUILD_VERSION}"
+      CLEAN_DIRECT_OUTPUT 1)
+  endif() #VERSIONED_OUTPUT
 elseif(BUILD_FRAMEWORKS_AND_BUNDLE)
   set_target_properties(${PROJ_CORE_TARGET}
     PROPERTIES
