diff -burN proj-6.2.1_orig/CMakeLists.txt proj-6.2.1/CMakeLists.txt
--- proj-6.2.1_orig/CMakeLists.txt	2019-10-28 18:49:32.000000000 +0100
+++ proj-6.2.1/CMakeLists.txt	2019-11-08 10:52:28.948109756 +0100
@@ -98,6 +98,7 @@
 proj_version(MAJOR 6 MINOR 2 PATCH 1)
 set(PROJ_API_VERSION "17")
 set(PROJ_BUILD_VERSION "17.1.2")
+option(VERSIONED_OUTPUT "suffix version into output name of binaries." ON)
 
 ################################################################################
 # Build features and variants
