diff -burN boost_1_60_0.orig/tools/build/src/tools/darwin.jam boost_1_60_0/tools/build/src/tools/darwin.jam
--- boost_1_60_0.orig/tools/build/src/tools/darwin.jam	2016-10-11 14:27:28.000000000 +0200
+++ boost_1_60_0/tools/build/src/tools/darwin.jam	2016-10-11 14:27:40.000000000 +0200
@@ -579,7 +579,7 @@
 
 actions link.dll bind LIBRARIES
 {
-    "$(CONFIG_COMMAND)" -dynamiclib -Wl,-single_module -install_name "$(<:B)$(<:S)" -L"$(LINKPATH)" -o "$(<)" "$(>)" "$(LIBRARIES)" -l$(FINDLIBS-SA) -l$(FINDLIBS-ST) $(FRAMEWORK_PATH) -framework$(_)$(FRAMEWORK:D=:S=) $(OPTIONS) $(USER_OPTIONS)
+    "$(CONFIG_COMMAND)" -dynamiclib -Wl,-single_module -install_name "@rpath/$(<:B)$(<:S)" -L"$(LINKPATH)" -o "$(<)" "$(>)" "$(LIBRARIES)" -l$(FINDLIBS-SA) -l$(FINDLIBS-ST) $(FRAMEWORK_PATH) -framework$(_)$(FRAMEWORK:D=:S=) $(OPTIONS) $(USER_OPTIONS)
 }
 
 # We use libtool instead of ar to support universal binary linking
