--- gdal-1.11.5-orig/ogr/ogrsf_frmts/kml/kml.cpp	Fri Jul  1 10:55:04 2016
+++ gdal-1.11.5/ogr/ogrsf_frmts/kml/kml.cpp	Thu Jul 18 13:58:27 2019
@@ -210 +210 @@
-    Attribute* poAtt = NULL;
+    KMLAttribute* poAtt = NULL;
@@ -234 +234 @@
-            poAtt = new Attribute();
+            poAtt = new KMLAttribute();
--- gdal-1.11.5-orig/ogr/ogrsf_frmts/kml/kmlnode.cpp	Fri Jul  1 10:55:04 2016
+++ gdal-1.11.5/ogr/ogrsf_frmts/kml/kmlnode.cpp	Thu Jul 18 14:00:21 2019
@@ -118 +118 @@
-    pvoAttributes_ = new std::vector<Attribute*>;
+    pvoAttributes_ = new std::vector<KMLAttribute*>;
@@ -369 +369 @@
-void KMLNode::addAttribute(Attribute *poAttr)
+void KMLNode::addAttribute(KMLAttribute *poAttr)
--- gdal-1.11.5-orig/ogr/ogrsf_frmts/kml/kmlnode.h	Fri Jul  1 10:55:04 2016
+++ gdal-1.11.5/ogr/ogrsf_frmts/kml/kmlnode.h	Thu Jul 18 13:58:57 2019
@@ -63 +63 @@
-    void addAttribute(Attribute* poAttr);
+    void addAttribute(KMLAttribute* poAttr);
@@ -100 +100 @@
-    typedef std::vector<Attribute*> kml_attributes_t;
+    typedef std::vector<KMLAttribute*> kml_attributes_t;
--- gdal-1.11.5-orig/ogr/ogrsf_frmts/kml/kmlutility.h	Fri Jul  1 10:55:04 2016
+++ gdal-1.11.5/ogr/ogrsf_frmts/kml/kmlutility.h	Thu Jul 18 13:59:22 2019
@@ -42 +42 @@
-struct Attribute
+struct KMLAttribute
