--- gdal-1.11.5-orig/port/cpl_vsil_win32.cpp	Fri Jul  1 10:55:03 2016
+++ gdal-1.11.5/port/cpl_vsil_win32.cpp	Wed Jul 17 16:41:35 2019
@@ -39 +39,2 @@
-
+#include <tchar.h>
+#include <stdio.h>
@@ -610,2 +611,2 @@
-        struct _wfinddata_t c_file;
-        intptr_t hFile;
+        WIN32_FIND_DATAW c_file;
+        HANDLE hFile = INVALID_HANDLE_VALUE;
@@ -622 +623 @@
-        if ( (hFile = _wfindfirst( pwszFileSpec, &c_file )) != -1L )
+        if ( (hFile = FindFirstFileW( pwszFileSpec, &c_file )) != INVALID_HANDLE_VALUE )			
@@ -627,2 +628,2 @@
-                    CPLRecodeFromWChar(c_file.name,CPL_ENC_UCS2,CPL_ENC_UTF8));
-            } while( _wfindnext( hFile, &c_file ) == 0 );
+                    CPLRecodeFromWChar(c_file.cFileName,CPL_ENC_UCS2,CPL_ENC_UTF8));
+            } while( FindNextFileW( hFile, &c_file ) != 0 );
@@ -630 +631 @@
-            _findclose( hFile );
+            FindClose( hFile );
@@ -647,2 +648,2 @@
-        struct _finddata_t c_file;
-        intptr_t hFile;
+        WIN32_FIND_DATA c_file;
+        HANDLE hFile = INVALID_HANDLE_VALUE;
@@ -657 +658 @@
-        if ( (hFile = _findfirst( pszFileSpec, &c_file )) != -1L )
+		if ( (hFile = FindFirstFile( pszFileSpec, &c_file )) != INVALID_HANDLE_VALUE )
@@ -661,2 +662,2 @@
-                oDir.AddString(c_file.name);
-            } while( _findnext( hFile, &c_file ) == 0 );
+				oDir.AddString(c_file.cFileName);
+            } while( FindNextFile( hFile, &c_file ) != 0 );
@@ -664 +665 @@
-            _findclose( hFile );
+            FindClose( hFile );
