--- /tmp/gdal-1.11.2/swig/python/GNUmakefile	2015-02-10 13:12:08.000000000 +0100
+++ /home/rkanavath/work/otb/source/SuperBuild/patches/GDAL/GNUmakefile	2016-05-10 15:32:17.934058573 +0200
@@ -68,7 +68,7 @@
 install:
 
 ifeq ($(PY_HAVE_SETUPTOOLS),1)
-	$(PYTHON) setup.py install 
+	$(PYTHON) setup.py install --prefix=$(DESTDIR)$(prefix) --install-purelib=$(prefix)/lib --install-scripts=$(prefix)/bin --install-platlib=$(prefix)/lib
 else
 	$(PYTHON) setup.py install --prefix=$(DESTDIR)$(prefix)
 endif
