--- gdal-3.4.0_orig/nmake.opt	2021-11-04 12:55:37.000000000 +0100
+++ gdal-3.4.0/nmake.opt	2021-11-22 16:22:46.247741400 +0100
@@ -307,9 +307,9 @@
 # This will enable 12bit libjpeg, using internal IJG libjpeg-6b in 12-bit mode
 # Note: it is possible to have 12bit JPEG support with this internal IJG libjpeg-6b
 # and 8bit JPEG support through an external libjpeg, such as libjpeg-turbo
-!IFNDEF JPEG12_SUPPORTED
-JPEG12_SUPPORTED = 1
-!ENDIF
+#!IFNDEF JPEG12_SUPPORTED
+#JPEG12_SUPPORTED = 1
+#!ENDIF
 
 #if using an external jpeg library uncomment the following lines
 # libjpeg-turbo is also supported (and recommended for better performance)
