--- gdal-1.11.5-orig/swig/makefile.vc	Fri Jul  1 10:55:08 2016
+++ gdal-1.11.5/swig/makefile.vc	Wed Jul 17 16:16:40 2019
@@ -21,0 +22,5 @@
+python-install: python
+        cd python
+        $(PYDIR)\python.exe setup.py install --prefix=$(GDAL_HOME)
+        # TODO: install python scripts
+
--- gdal-1.11.5-orig/makefile.vc	Fri Jul  1 10:55:03 2016
+++ gdal-1.11.5/makefile.vc	Wed Jul 17 16:16:04 2019
@@ -61 +61,9 @@
-DISTDIR	=	gdal_$(VERSION)
+DISTDIR	= gdal_$(VERSION)
+
+WRAP_TARGETS =
+
+!IFDEF WRAP_PYTHON
+WRAP_TARGETS = $(WRAP_TARGETS) swig_py
+!ENDIF
+
+
@@ -64 +72 @@
-		plugin_dir apps_dir 
+		plugin_dir apps_dir  $(WRAP_TARGETS)
@@ -172,0 +181,5 @@
+swig_py:   $(TARGET_LIB)
+  cd swig
+  $(MAKE) /f makefile.vc python
+  cd ..
+
@@ -199,0 +213,6 @@
+!ENDIF
+
+!IFDEF WRAP_PYTHON
+	cd swig
+	$(MAKE) /f makefile.vc python-install
+	cd ..
