--- gdal-2.4.1_orig/frmts/hdf5/makefile.vc	2019-03-15 13:30:02.000000000 +0100
+++ gdal-2.4.1/frmts/hdf5/makefile.vc	2019-04-05 17:43:22.356322797 +0200
@@ -7,7 +7,7 @@
 
 PLUGIN_DLL 	=	gdal_HDF5.dll
 
-EXTRAFLAGS 	= 	-I$(HDF5_DIR)\include -DWIN32 -D_HDF5USEDLL_ 
+EXTRAFLAGS 	= 	-I$(HDF5_DIR)\include -DWIN32 -DH5_BUILT_AS_DYNAMIC_LIB 
 
 !IF "$(HDF5_PLUGIN)" == "YES"
 EXTRAFLAGS = $(EXTRAFLAGS) -DHDF5_PLUGIN
