--- gdal-1.11.5.orig/frmts/raw/envidataset.cpp	Fri Jul  1 10:55:06 2016
+++ gdal-1.11.5/frmts/raw/envidataset.cpp	Wed Jul 17 16:12:44 2019
@@ -1500,4 +1500,8 @@
     if( oSRS.GetRoot() == NULL )
-        oSRS.SetLocalCS( papszFields[0] );
-
+        {
+         oSRS.SetLocalCS(papszFields[0]);
+        // assume that the Y axis isn't flipped
+        adfGeoTransform[2] *= -1.0;
+        adfGeoTransform[5] *= -1.0;
+        }
 /* -------------------------------------------------------------------- */
