diff -burN qwt-6.1.3-orig/qwtbuild.pri qwt-6.1.3/qwtbuild.pri
--- qwt-6.1.3-orig/qwtbuild.pri	2017-09-07 15:00:11.755801441 +0200
+++ qwt-6.1.3/qwtbuild.pri	2017-09-07 15:00:55.812189641 +0200
@@ -21,14 +21,8 @@
 ######################################################################
 
 win32 {
-    # On Windows you can't mix release and debug libraries.
-    # The designer is built in release mode. If you like to use it
-    # you need a release version. For your own application development you
-    # might need a debug version.
-    # Enable debug_and_release + build_all if you want to build both.
 
-    CONFIG           += debug_and_release
-    CONFIG           += build_all
+    CONFIG           += release
 }
 else {
 
