diff -burN Shark-349f29bd71c370e0f88f7fc9aa66fa5c4768fcb0.orig/CMakeLists.txt Shark-349f29bd71c370e0f88f7fc9aa66fa5c4768fcb0/CMakeLists.txt
--- Shark-349f29bd71c370e0f88f7fc9aa66fa5c4768fcb0.orig/CMakeLists.txt	2017-08-22 11:31:50.472052695 +0200
+++ Shark-349f29bd71c370e0f88f7fc9aa66fa5c4768fcb0/CMakeLists.txt	2017-08-22 11:32:36.448358789 +0200
@@ -141,10 +141,8 @@
 
 find_package( 
 	Boost 1.48.0 REQUIRED COMPONENTS
-	system date_time filesystem
-	program_options serialization thread
-	unit_test_framework
-)
+	serialization
+	)
 
 if(NOT Boost_FOUND)
 	message(FATAL_ERROR "Please make sure Boost 1.48.0 is installed on your system")
